### PR TITLE
GRO-10: Improve artist canonical link computation

### PR DESCRIPTION
**Pull request by jonallured**
Trying this again from a feature branch.

https://artsyproduct.atlassian.net/browse/GRO-10

### CODE REVIEW

**Comment by jonallured** [comment]
Hey @eessex you had this comment on the original PR:

https://github.com/artsy/force/pull/6478#discussion_r504841780

And I didn't want to lose that note - but I'm not sure what you mean! Can you give me a nudge about what you'd like to see change here?

**Comment by eessex** [comment]
@jonallured we recently added some helpers to consolidate how we are grabbing page slug and type on the client, it has a return value like so:
Screen Shot 2020-10-14 at 3 24 54 PM

It seems like canonical path information could easily live in this helper, and it might already have some of the info you need... this will allow you to avoid passing `path` down through props.

**Comment by jonallured** [comment]
@eessex gotcha so you're envisioning a change to that helper that would allow it to return a shape like this:

```ts
{
  pageCanonical: "",
  pageParts: [],
  pageSlug: "",
  pageType: ""
}
```

So then at the call site we'd see this:

```ts
export const ArtistMetaCanonicalLink: React.FC<ArtistMetaCanonicalLinkProps> = ({ artist }) => {
  const { pageCanonical } = getPageTypeFromClient()
  return <Link rel="canonical" href={pageCanonical} />
}
```

Did I get that right? This is interesting, my only concern is that I have at least one case where the canonical url depends on the shape of an artwork object. Would we pass this artwork data to the helper?

**Comment by eessex** [comment]
@jonallured yeah i think the change you suggested could work, and then update the canonical URL from the component if required. Non-blocking, but would be great be more consistent with how we generate paths across the site.

File: src/v2/Apps/Artist/Components/ArtistMetaCanonicalLink.tsx [hunk 1]
```diff
@@ -6,27 +6,43 @@ import { ArtistMetaCanonicalLink_artist } from "v2/__generated__/ArtistMetaCanon
...
 }
 
 export type ArtistMetaCanonicalLinkProps = {
   artist: ArtistMetaCanonicalLink_artist
+  path: string
 }
 
 export const ArtistMetaCanonicalLink: React.FC<ArtistMetaCanonicalLinkProps> = ({
   artist,
+  path,
```
Comment (damassi) [inline src/v2/Apps/Artist/Components/ArtistMetaCanonicalLink.tsx:41]: No need to pass props as [we can use our hook](https://github.com/artsy/force/blob/master/src/v2/Artsy/Router/useRouter.tsx)

**Review by damassi — APPROVED** [review]
One requested change, but after that's done LGTM 👍

**Comment by damassi** [comment]
EDIT: sorry, didn't see @eessex 's comments!

**Comment by jonallured** [comment]
Ok @damassi how does 2698e16 look - is that what you were expecting?

I think I'm going to punt a little here on moving this computation up to the `getPageTypeFromClient` helper as we actually have a ticket for a larger refactor that I'd like to defer this work to:

https://artsyproduct.atlassian.net/browse/GRO-15

I guess what I'm thinking is that it's worth shipping the improvement here and then we can come back to this topic when we hit that larger ticket. At least we'll get the duplicated `canonical` link fixed and that seems worth doing!

**Comment by damassi** [comment]
Looks good @jonallured 👍 👍

**Comment by jonallured** [comment]
Ok going to merge and follow up on the next ticket in this line of work for the refactor you mentioned @eessex!!

**Pull request merged**